File: pytools/data/data.code.golang.client/req_role_login.go
package proto

import (
	"packet"
)

type ReqRoleLogin struct {
	uid                      uint32
	uuid                     uint32
	sid                      uint16
	cid                      uint16
	loginTime                uint32
	pwd                      string
	relink                   uint8
	debug                    uint8
	os                       string
	version                  string
}

func (this *ReqRoleLogin) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.uid)
	pack.WriteUint32(this.uuid)
	pack.WriteUint16(this.sid)
	pack.WriteUint16(this.cid)
	pack.WriteUint32(this.loginTime)
	pack.WriteString(this.pwd)
	pack.WriteUint8(this.relink)
	pack.WriteUint8(this.debug)
	pack.WriteString(this.os)
	pack.WriteString(this.version)

	return pack.Encode(P_REQ_ROLE_LOGIN)
}

func (this *ReqRoleLogin) SetUid(uid uint32) {
	this.uid = uid
}

func (this *ReqRoleLogin) SetUuid(uuid uint32) {
	this.uuid = uuid
}

func (this *ReqRoleLogin) SetSid(sid uint16) {
	this.sid = sid
}

func (this *ReqRoleLogin) SetCid(cid uint16) {
	this.cid = cid
}

func (this *ReqRoleLogin) SetLoginTime(loginTime uint32) {
	this.loginTime = loginTime
}

func (this *ReqRoleLogin) SetPwd(pwd string) {
	this.pwd = pwd
}

func (this *ReqRoleLogin) SetRelink(relink uint8) {
	this.relink = relink
}

func (this *ReqRoleLogin) SetDebug(debug uint8) {
	this.debug = debug
}

func (this *ReqRoleLogin) SetOs(os string) {
	this.os = os
}

func (this *ReqRoleLogin) SetVersion(version string) {
	this.version = version
}

File: pytools/data/data.code.lua.client/ReqRoleLogin.lua
module("ReqRoleLogin", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqRoleLogin})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uid = 0
	self.uuid = 0
	self.sid = 0
	self.cid = 0
	self.login_time = 0
	self.pwd = 0
	self.relink = 0
	self.debug = 0
	self.os = 0
	self.version = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.uid)
	pack:writeU32(self.uuid)
	pack:writeU16(self.sid)
	pack:writeU16(self.cid)
	pack:writeU32(self.login_time)
	pack:writeString(self.pwd)
	pack:writeU8(self.relink)
	pack:writeU8(self.debug)
	pack:writeString(self.os)
	pack:writeString(self.version)
	return pack:encode(Msg.P_REQ_ROLE_LOGIN)
end


function setUid(self, uid)
	self.uid = uid
end

function setUuid(self, uuid)
	self.uuid = uuid
end

function setSid(self, sid)
	self.sid = sid
end

function setCid(self, cid)
	self.cid = cid
end

function setLoginTime(self, login_time)
	self.login_time = login_time
end

function setPwd(self, pwd)
	self.pwd = pwd
end

function setRelink(self, relink)
	self.relink = relink
end

function setDebug(self, debug)
	self.debug = debug
end

function setOs(self, os)
	self.os = os
end

function setVersion(self, version)
	self.version = version
end

